python/cudf/cudf/core/dataframe.py

```diff
@@ -785,7 +785,9 @@ def empty(self):
         out : bool
             If DataFrame is empty, return True, if not return False.
         """
-        return not len(self)
+        if self._num_columns == 0:
+            return True
+        return self._num_rows == 0
```

Comment:
Should we just define this at the `Frame` level so that `Series` and `Index` objects also inherit the `empty` method?

Reply:
Pushed the core-logic to `Frame.empty` but had to keep the Class level properties for the sake of documentation.